org.adempiere.base/src/org/compiere/acct/Doc_AllocationHdr.java

```diff
@@ -460,6 +466,7 @@ else if (isUsingClearing && line.getC_CashLine_ID() != 0) // Avoid usage of clea
 				&& (getC_Currency_ID() != as.getC_Currency_ID()			//	payment allocation in foreign currency
 					|| getC_Currency_ID() != line.getInvoiceC_Currency_ID()))	//	allocation <> invoice currency
 			{
+				//do need a check here for as.isAccrual()?
```

Comment:
@hengsin @CarlosRuiz-globalqss 
wdyt about this check? just uncheck "Accrual" on "Accounting Schema" and run unit test for allocation some not pass by NPE at createInvoiceGainLoss

![image](https://github.com/idempiere/idempiere/assets/4666285/80f974ab-8c3f-412c-812f-b6f12bba03ab)

Reply:
Well, seems like a different bug, not to mix with this ticket.

Cash based accounting (non-accrual) is rarely used, that must be the reason nobody has caught such error.

Reply:
Ah @hieplq - also answering your question (I didn't notice at first) - most tests are designed to work in a vanilla seed GardenWorld - some tests will fail if the GW configuration is modified, or even if there is additional data.

Not sure if that's something we must solve generically in all tests, but sounds like big work.

Reply:
hi @hieplq , that test is for accrual accounting only. I guess the fix is pretty simple for this one, we just have to either make sure accounting schema is accrual accounting or skip the test if accounting schema is non-accrual.